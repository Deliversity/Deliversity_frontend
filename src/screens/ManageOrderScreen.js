import React, {Component} from 'react';
import {Text, View, StyleSheet, FlatList, RefreshControl} from 'react-native';
import axios from '../axiosConfig';
import Card from '../components/manageOrderCard';
class ManageOrderScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      orderInfo: '',
      orderId: [], ///orderList
      refreshing: false,
    };
  }
  componentDidMount() {
    this.getOrderInfo();
  }
  getOrderInfo = async () => {
    //console.log(this.state.orderID);
    await this.setState({refreshing: true});
    const res = await axios.get('/api/v1/order/orderList');
    await this.setState({refreshing: false, orderInfo: res.data.data});
    //return res.data.data;
  };
  handleRefresh = async () => {
    await this.getOrderInfo();
  };
  renderItem = ({item}) => {
    return (
      <Card
        itemData={item}
        onPress={() => {
          item.orderStatus === '3'
            ? this.props.navigation.navigate('WriteReview', {
                orderID: item.id,
                riderID: item.riderId,
              })
            : this.props.navigation.navigate('DeliveryMan', {
                orderID: item.id,
              });
        }}
      />
    );
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.text_header}>나의 주문 리스트</Text>
        </View>
        <View style={styles.footer}>
          <FlatList
            refreshControl={
              <RefreshControl
                refreshing={this.state.refreshing}
                onRefresh={this.handleRefresh}
              />
            }
            extraData={this.state}
            data={this.state.orderInfo}
            renderItem={this.renderItem}
            keyExtractor={(item) => item.id}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#8fbc8f',
  },
  header: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text_header: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 20,
  },
  footer: {
    flex: 8,
    backgroundColor: '#fff',
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30,
    paddingVertical: 20,
  },
  textSize: {
    fontWeight: 'bold',
    fontSize: 17,
  },
  bookingStyle: {
    fontSize: 13,
    color: '#ff7f50',
  },
  reservationBox: {
    borderWidth: 2,
    borderColor: '#ff7f50',
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    borderBottomLeftRadius: 15,
    borderBottomRightRadius: 15,
    paddingHorizontal: 10,
  },
});

export default ManageOrderScreen;
