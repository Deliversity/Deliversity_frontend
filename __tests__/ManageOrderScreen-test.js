import 'react-native';

import React, {ReactElement} from 'react';
import ManageOrderScreen from '../src/screens/OrderManage/ManageOrderScreen';
import renderer from 'react-test-renderer';

let props;
let component: ReactElement;

function getTempComponent(props) {
  return <ManageOrderScreen {...props} />;
}

beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
});

describe('[Temp] render', () => {
  props = {}; // fill test props
  component = getTempComponent(props);
  it('renders without crashing', () => {
    const renderResult = renderer.create(component);
    const renderedJson = renderResult.toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
