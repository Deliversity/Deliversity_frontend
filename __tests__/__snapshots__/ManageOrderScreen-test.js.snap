// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#8fbc8f",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      나의 주문 리스트
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 8,
        "paddingVertical": 20,
      }
    }
  >
    <RCTScrollView
      data=""
      disableVirtualization={false}
      extraData={
        Object {
          "orderId": Array [],
          "orderInfo": "",
          "refreshing": true,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={true}
        />
      }
      removeClippedSubviews={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
